src/PIL/PngImagePlugin.py

```diff
@@ -216,15 +216,20 @@ class PngInfo:
     def __init__(self):
         self.chunks = []
 
-    def add(self, cid, data):
+    def add(self, cid, data, after_idat=False):
         """Appends an arbitrary chunk. Use with caution.
 
         :param cid: a byte string, 4 bytes long.
         :param data: a byte string of the encoded data
+        :param after_idat: for use with private chunks. Whether the chunk
+                           should be written after IDAT
 
         """
 
-        self.chunks.append((cid, data))
+        chunk = [cid, data]
+        if after_idat:
+            chunk.append(True)
+        self.chunks.append(tuple(chunk))
```

Comment:
Code would simpler if you always save the flag.

```suggestion
        self.chunks.append((cid, data, after_idat))
```

Reply:
My thinking was that this is more backwards compatible.

If someone had this code before -

```python
from PIL import PngImagePlugin
info = PngImagePlugin.PngInfo()
info.add("priV", "\x00")
cid, data = info.chunks[0]
```

Then your suggestion would break it with `ValueError: too many values to unpack`. I will say that using the class in this way doesn't seem likely however.

Reply:
Since `PngInfo.chunks` is undocumented, I think we can sacrifice backward compatibility for convenience.